core/src/main/java/org/sonatype/goodies/dropwizard/util/FileHelper.java

```diff
@@ -227,4 +228,34 @@ public static void createDirectories(final Path dir) throws IOException {
       checkState(Files.isWritable(dir), "Directory not writable: %s", dir);
     }
   }
+
+  /**
+   * Create directories.
+   *
+   * @since ???
+   */
+  public static void createDirectories(final File dir) throws IOException {
+    checkNotNull(dir);
+    createDirectories(dir.toPath());
+  }
+
+  /**
+   * Create a new file reference; checking for directory traversal.
+   *
+   * Caller should ensure that basedir has already been made canonical.
+   *
+   * @since ???
+   */
+  @Beta
+  public static File newFile(final File basedir, final String path) {
+    checkNotNull(basedir);
+    checkNotNull(path);
+    log.trace("New file: {}/{}", basedir, path);
+
+    File file = new File(basedir, path).toPath().normalize().toFile();
+    checkState(file.getPath().startsWith(basedir.getPath()),
```

Comment:
We need `Path.startsWith()` instead of `String.startsWith()` to cover all cases, i.e.
`file.toPath().startsWith(basedir.toPath())`

Otherwise, `basedir=/foo` and `path=../foobar` get past the check.